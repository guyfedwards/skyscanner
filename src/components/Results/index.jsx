import React from 'react';
import propTypes from 'prop-types';

import flightType from '../../types/flight';
import Result from '../Result';

const Results = ({ flights }) =>
  flights.map(f => <Result flight={f} key={f.id} />);

Results.propTypes = {
  flights: propTypes.arrayOf(flightType).isRequired,
};

export default Results;
