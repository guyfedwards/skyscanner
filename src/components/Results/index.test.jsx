import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-test-renderer/shallow';

import Results from '.';

const flights = [
  {
    id: 'it_1',
    legs: [
      {
        id: 'leg_1',
        departure_airport: 'BUD',
        arrival_airport: 'LTN',
        departure_time: '2020-10-31T15:35',
        arrival_time: '2020-10-31T17:00',
        stops: 0,
        airline_name: 'Wizz Air',
        airline_id: 'WZ',
        duration_mins: 145,
      },
    ],
    price: 'Â£35',
    agent: 'Wizzair.com',
    agent_rating: 9.1,
  },
];

describe('Results', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<Results flights={flights} />, div);
  });

  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Results flights={flights} />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
