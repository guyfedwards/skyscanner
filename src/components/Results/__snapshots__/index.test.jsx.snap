// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should render correctly 1`] = `
Array [
  <Result
    flight={
      Object {
        "agent": "Wizzair.com",
        "agent_rating": 9.1,
        "id": "it_1",
        "legs": Array [
          Object {
            "airline_id": "WZ",
            "airline_name": "Wizz Air",
            "arrival_airport": "LTN",
            "arrival_time": "2020-10-31T17:00",
            "departure_airport": "BUD",
            "departure_time": "2020-10-31T15:35",
            "duration_mins": 145,
            "id": "leg_1",
            "stops": 0,
          },
        ],
        "price": "Â£35",
      }
    }
  />,
]
`;
